cmd/gardenlet/app/migration.go

```diff
@@ -165,3 +176,41 @@ func syncBackupSecretRefAndCredentialsRef(backup *gardencorev1beta1.Backup) {
 	// - both fields are set -> let the validation check if they are correct
 	// - credentialsRef refer to WorkloadIdentity -> secretRef should stay unset
 }
+
+// TODO(@aaronfern): Remove this after v1.122 is released
+func migrateMCMRBAC(ctx context.Context, seedClient client.Client) error {
+	namespaceList := &corev1.NamespaceList{}
+	if err := seedClient.List(ctx, namespaceList, client.MatchingLabels(map[string]string{v1beta1constants.GardenRole: v1beta1constants.GardenRoleShoot})); err != nil {
+		return err
+	}
+
+	var tasks []flow.TaskFn
+
+	for _, namespace := range namespaceList.Items {
+		if namespace.DeletionTimestamp != nil || namespace.Status.Phase == corev1.NamespaceTerminating {
+			continue
+		}
+		tasks = append(tasks, func(ctx context.Context) error {
+			clusterRoleBinding := &rbacv1.ClusterRoleBinding{}
+			if err := seedClient.Get(ctx, client.ObjectKey{Name: "machine-controller-manager-" + namespace.Name}, clusterRoleBinding); err != nil {
+				//If MCM clusterRoleBinding does not exist, nothing to do
+				if apierrors.IsNotFound(err) {
```

Comment:
Nit: When the error is not nil, you can directly return here with:
```suggestion
				return client.IgnoreNotFound(err)
```